util/stats/query_stats.go

```diff
@@ -111,7 +122,15 @@ func NewQueryStats(tg *QueryTimers) *QueryStats {
 		}
 	}
 
-	qs := QueryStats{Timings: qt}
+	if ts != nil {
+		samples = querySamples{
+			EvaluatorCount: ts.EvaluatorCount,
+			PeakSamples:    ts.PeakSamples,
+			TotalSamples:   ts.TotalSamples,
+		}
+	}
+
+	qs := QueryStats{Timings: qt, Samples: samples}
```

Comment:
```suggestion
	qs := QueryStats{Timings: qt, Samples: ts}
```

alwo why ts  as var name?

Reply:
yeah it shouldn't be tha tway. I have now changed it to `sp` for `samples`